### PR TITLE
Add Merged PRs, fetch by query

### PR DESCRIPTION
**Pull request by shubhsherl**
#20 
Fetching user data by the query, and populating initial user list using repo contributions, issues, PRs.
Initial fetching may take upto 1hr, but usual refresh will take at the rate of 10users/minute.(Since API allows only 30 requests per minute and each user req 3 API calls).
To update the repo and populating List of all users we can run `python manage.py fetch`, usual refresh will be done by `python manage.py refresh`.

### CODE REVIEW

File: refresh.sh [hunk 1]
```diff
@@ -1,6 +1,10 @@
 #!/bin/sh  
 while true  
 do
+if [ 1700 -lt `date +%H%M` -a `date +%H%M` -lt 1710 ]
```
Comment (shubhsherl) [inline refresh.sh]: To fetch the full list once a day. Typically takes an hour.

**Comment by Sing-Li** [comment]
@shubhsherl   How long does the "refresh" without the "fetch full list" take?   And does it vary with the number of GSoC students?   If it does, what is the formula for total time?

**Comment by shubhsherl** [comment]
@Sing-Li  Usual refresh(only GSoC list) will run at the rate of 10users/minute.

File: core/views.py [hunk 2]
```diff
@@ -102,93 +109,93 @@ def getRepoIssues(owner, repoName, url=''):
...
-                        totalIssues=contributors[username]['issue_counts'])
-                else:
-                    newUser = User(
-                        login=username, avatar=issue['user']['avatar_url'], totalIssues=contributors[username]['issue_counts'])
-                    newUser.save()
+        username = issue['user']['login'].lower()
+        if not username in contributors:
+            contributors[username] = {}
+            contributors[username]['updated'] = True
+            if not User.objects.filter(login=username):
```
Comment (kareemhamdy100) [inline core/views.py]: access database slowdown the code

File: core/views.py [hunk 3]
```diff
@@ -102,93 +109,93 @@ def getRepoIssues(owner, repoName, url=''):
...
-    return contributors
-
-
-def saveIssues(issues, contributors_):
-    contributors = contributors_
+        username = pull['user']['login'].lower()
+        if not username in contributors:
+            contributors[username] = {}
+            contributors[username]['updated'] = True
+            if not User.objects.filter(login=username):
```
Comment (kareemhamdy100) [inline core/views.py]: access database slowdown the code

File: core/views.py [hunk 4]
```diff
@@ -102,93 +109,93 @@ def getRepoIssues(owner, repoName, url=''):
...
-            if User.objects.filter(login=username):
-                User.objects.filter(login=username).update(
-                    totalCommits=contributors[username]['commits'])
-        else:
+        if not username in contributors:
             contributors[username] = {}
-            contributors[username]['commits'] = commits
-            contributors[username]['PR_counts'] = 0
-            contributors[username]['issue_counts'] = 0
+            contributors[username]['updated'] = True
```
Comment (kareemhamdy100) [inline core/views.py]: access database slowdown the code

File: core/views.py [hunk 5]
```diff
@@ -102,93 +109,93 @@ def getRepoIssues(owner, repoName, url=''):
...
+                newUser.save()
+            update(username)
     return contributors
 
+def update(username):
+    getMergedPRs(username)
+    getIssues(username)
+    getOpenPRs(username)
+
+def getMergedPRs(username):
```
Comment (kareemhamdy100) [inline core/views.py:30]: not needed and slow

File: core/views.py [hunk 6]
```diff
@@ -102,93 +109,93 @@ def getRepoIssues(owner, repoName, url=''):
...
+        mergedPRs = response.json()['total_count']
+        if User.objects.filter(login=username):
+            User.objects.filter(login=username).update(totalMergedPRs=mergedPRs)
+    elif (response.status_code == 403): # rate-limit exceeded wait for 30sec 
+        time.sleep(30)
+        getMergedPRs(username)        
+    return mergedPRs != -1
+
+def getIssues(username):
+    issues = -1
```
Comment (kareemhamdy100) [inline core/views.py:46]: not needed and slow

File: core/views.py [hunk 7]
```diff
@@ -102,93 +109,93 @@ def getRepoIssues(owner, repoName, url=''):
...
+    if (response.status_code == 200):
+        issues = response.json()['total_count']
+        if User.objects.filter(login=username):
+            User.objects.filter(login=username).update(totalIssues=issues)
+    elif (response.status_code == 403):
+        time.sleep(30)
+        getIssues(username)
+    return issues != -1
+
+def getOpenPRs(username):
```
Comment (kareemhamdy100) [inline core/views.py:60]: not needed and slow

**Review by kareemhamdy100 — COMMENTED** [review]
you need to optimizecode that access database

**Pull request closed without merging**